### PR TITLE
Set password page

### PR DESCRIPTION
**Pull request by robbiecarlton**

### CODE REVIEW

**Comment by robbiecarlton** [comment]
@levity - would you take a look at this when you get the chance.

File: src/containers/user/UserSettings.js [hunk 1]
```diff
@@ -26,6 +26,8 @@ import Icon from '../../components/Icon'
 @prefetch(({ dispatch, params: { id }, query }) => {
   switch (query.expand) {
     case 'password':
+      dispatch(toggleUserSettingsSection('account', true))
+      return dispatch(toggleUserSettingsSection('password', true))
```
Comment (robbiecarlton) [inline src/containers/user/UserSettings.js:30]: reusing this action and store is a little hacky, but it also makes sense to me. Possibly the action needs renaming?

Comment (levity) [inline src/containers/user/UserSettings.js:30]: fine with me

File: src/containers/Login.js [hunk 2]
```diff
@@ -107,6 +107,9 @@ export default class Login extends React.Component {
 
     if (error === 'no user') error = 'Login was canceled or no user data was found.'
     if (error === 'no email') error = 'The user data did not include an email address.'
+    if (error && error.match(/password account not found. available: \[(.*)\]/)) {
+      error = <span>Your account has no password set. <a href='/set-password'>Set your password</a></span>
+    }
```
Comment (levity) [inline src/containers/Login.js]: can you parse the "available" section and display messages like "Or use Facebook" as appropriate, so this doesn't confuse people who use third-party auth?

**Comment by levity** [comment]
add screenshots of the page on desktop & mobile, please

**Comment by robbiecarlton** [comment]
Will do. It'll have to be tomorrow morning at this point.

On Wednesday, May 18, 2016, Lawrence Wang notifications@github.com wrote:

> add screenshots of the page on desktop & mobile, please
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/Hylozoic/hylo-redux/pull/54#issuecomment-220176553

##

**Comment by robbiecarlton** [comment]
screen shot 2016-05-19 at 11 34 18 am
screen shot 2016-05-19 at 11 34 27 am
screen shot 2016-05-19 at 11 34 40 am
screen shot 2016-05-19 at 11 34 53 am

**Comment by robbiecarlton** [comment]
@levity - good to go?

**Comment by levity** [comment]
:+1:

**Pull request merged**